catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/monitor/impl/WorkspaceServiceImpl.java

```diff
@@ -169,11 +179,22 @@ private QueryResponse query(QueryRequest queryRequest) {
     return workspaceMetacard
         .getQueries()
         .stream()
-        .map(workspaceTransformer::xmlToMetacard)
+        .map(this::getAssociatedMetacardById)
```

Comment:
❗️We should use `endpointUtil.getMetacards(ids, "query")` to retrieve the query metacards so we don't have to send a separate query for each.